src/app/games/trexo/TrexoMinimax.ts

```diff
@@ -0,0 +1,46 @@
+import { BoardValue } from 'src/app/jscaip/BoardValue';
+import { Coord } from 'src/app/jscaip/Coord';
+import { MGPNode } from 'src/app/jscaip/MGPNode';
+import { Minimax } from 'src/app/jscaip/Minimax';
+import { Player, PlayerOrNone } from 'src/app/jscaip/Player';
+import { SCORE } from 'src/app/jscaip/SCORE';
+import { TrexoMove } from './TrexoMove';
+import { TrexoNode, TrexoRules } from './TrexoRules';
+import { TrexoState } from './TrexoState';
+
+export class TrexoMinimax extends Minimax<TrexoMove, TrexoState> {
+
+    public getListMoves(node: TrexoNode): TrexoMove[] {
+        return TrexoRules.get().getLegalMoves(node.gameState);
+    }
+    public getBoardValue(node: TrexoNode): BoardValue {
+        let score: number = 0;
+        const state: TrexoState = node.gameState;
+        const lastPlayer: Player = state.getCurrentOpponent();
+        let lastPlayerAligned5: boolean = false;
+        for (let x: number = 0; x < TrexoState.SIZE; x++) {
+            // for every column, starting from the bottom of each column
+            for (let y: number = 0; y < TrexoState.SIZE; y++) {
+                // while we haven't reached the top or an empty space
+                const pieceOwner: PlayerOrNone = state.getPieceAtXY(x, y).getOwner();
+                if (pieceOwner.isPlayer()) {
+                    const tmpScore: number = TrexoRules.getSquareScore(state, new Coord(x, y));
+                    if (MGPNode.getScoreStatus(tmpScore) === SCORE.VICTORY) {
+                        if (pieceOwner === lastPlayer) {
+                            lastPlayerAligned5 = true;
```

Comment:
pourquoi pas retourner le board value direct ici ?

Reply:
Bah, c'est pas encore sur qu'il a gagné hein, ça dépend de si on découvre après que l'autre a aligné aussi !

Reply:
ok ça fait sens ! Tu peux ajouter un mini commentaire pour clarifier ça ? Et renommer lastPlayer en currentPlayer ?

Reply:
personne est le current player dans ce contexte ci, si ?
Mais c'est commencé.

Reply:
cause last player only win if the other don't align
->
because the last player only wins if the other does not get an alignment

Reply:
ok mister fancy pants !